### ISSUE TITLE
Scheduler should be able to save to worklist

### ISSUE THREAD

**Issue by resurl** [ubclaunchpad/life-at-ubc#59]
saving worklists was one of the things we discussed in ideation, but we couldn't find much of a solution back then.
i think it might be still doable, and this would give us a leg up on any other existing ubc course schedulers (i think?).

we cant implement this into the code right away, at least not until we get the MVP

**Comment by resurl** [comment]
this link ([https://courses.students.ubc.ca/cs/courseschedule?pname=subjarea&tname=subj-course&dept=AGEC&course=530A&submit=save](https://courses.students.ubc.ca/cs/courseschedule?pname=subjarea&tname=subj-course&dept=AGEC&course=530A&submit=save)) lets you save AGEC 530A to your active worklist (if logged in), so we can probably use that somehow?

[https://courses.students.ubc.ca/cs/courseschedule?pname=subjarea&tname=subj-section&dept=CPSC&course=110&section=101&submit=save](https://courses.students.ubc.ca/cs/courseschedule?pname=subjarea&tname=subj-section&dept=CPSC&course=110&section=101&submit=save) this lets you save CPSC 110 101 to your worklist

**Comment by johnagl** [comment]
Given that we've deployed the application, I think this is one **new** feature that is very doable for term 2 and a useful one at that.

**Comment by johnagl** [comment]
@resurreccionl I'm starting to think about this feature, since we're getting close to finishing some of the other polish tasks. For saving, how useful would it be to save the schedule on our application - but not on the worklist. 

A feature such as: 
1) User saves schedule and receives a link or code
2) User can paste link/code and their saved schedule shows up

This allows them to share the schedule - but... it's not on their worklist lol. I'm also not sure of the feasability of saving directly on worklist, we would have to look into that.

**Comment by resurl** [comment]
I think it might be impossible (??) to save multiple courses to worklist at once, unfortunately. The SSC has no API at all (🤮) and the only way to save a course is in this comment: 

> this link (https://courses.students.ubc.ca/cs/courseschedule?pname=subjarea&tname=subj-course&dept=AGEC&course=530A&submit=save) lets you save AGEC 530A to your active worklist (if logged in), so we can probably use that somehow?
> 
> https://courses.students.ubc.ca/cs/courseschedule?pname=subjarea&tname=subj-section&dept=CPSC&course=110&section=101&submit=save this lets you save CPSC 110 101 to your worklist

so we CAN save things to worklist, buuut it's kinda scuffed LOL. The worklist has to be already made and selected on the SSC (with the user logged in already), and then the user needs to either click each url to save to the worklist or we can give them a button and that button would click all the urls for them. Either way, i think that results in new tabs being opened for the user? which might be a bit weird lol

but i do think the linked schedule idea would be pretty nice to have, even without the saving feature. would it be possible to also implement editing saved schedules? i think if we had that, there'd be even more value in saving schedules (without saving).

**Comment by johnagl** [comment]
EDIT: This is a second option, a shareable link to a schedule.

Okay... yeah the saving to UBC worklist does seem pretty hard. I'm thinking saving a schedule would look something like:

1) In generate screen, have a save schedule button
2) Saving schedule should generate a link or id of some sort - this is stored in the db, along with the courses they selected
3) If user attempts to go to save url (eg. courseload.com/saved/:id) we check for the given id in the database, if it exists, we can display their schedule

Haven't thought about the editing schedule part yet, but I'm hoping we can just reuse the current code we have

**Comment by johnagl** [comment]
> I think it might be impossible (??) to save multiple courses to worklist at once, unfortunately. The SSC has no API at all (🤮) and the only way to save a course is in this comment:
> 
> > this link (https://courses.students.ubc.ca/cs/courseschedule?pname=subjarea&tname=subj-course&dept=AGEC&course=530A&submit=save) lets you save AGEC 530A to your active worklist (if logged in), so we can probably use that somehow?
> > https://courses.students.ubc.ca/cs/courseschedule?pname=subjarea&tname=subj-section&dept=CPSC&course=110&section=101&submit=save this lets you save CPSC 110 101 to your worklist
> 
> so we CAN save things to worklist, buuut it's kinda scuffed LOL. The worklist has to be already made and selected on the SSC (with the user logged in already), and then the user needs to either click each url to save to the worklist or we can give them a button and that button would click all the urls for them. Either way, i think that results in new tabs being opened for the user? which might be a bit weird lol
> 
> but i do think the linked schedule idea would be pretty nice to have, even without the saving feature. would it be possible to also implement editing saved schedules? i think if we had that, there'd be even more value in saving schedules (without saving).

Okay clearly I didn't read this thoroughly the first time - but I think this is still pretty useful and quite simple to implement.

Basically, if you click the links @resurreccionl provided, it automatically adds it to an existing worklist (as long as you're logged in). We can generate these links easily on the frontend..
